types.go

```diff
@@ -0,0 +1,127 @@
+package jwt
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+// TimePrecision sets the precision of times and dates within this library.
+// This has an influence on the precision of times when comparing expiry or
+// other related time fields. Furthermore, it is also the precision of times
+// when serializing.
+//
+// For backwards compatibility the default precision is set to seconds, so that
+// no fractional timestamps are generated.
+//
+// TODO(oxisto): the tests seem to fail sometimes, if the precision is microseconds because the difference is literally 1 microsecond
```

Comment:
Is there anything further you could pinpoint on this? from seeing the implementation I'd put my money on a IEEE rounding error but that should be reproducible and since it is a seemingly random microsecond error on the same test it likely is a clock skew

time.Unix doesn't involve any float aritmetic https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/time/time.go;l=1317-1328

the likely cause of the skew is time.Truncate, which does indeed some extensive calculations on the division and then strips the monotonic clock reading (time.ext) https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/time/time.go;l=1442-1453;drc=refs%2Ftags%2Fgo1.17. Additionally, time.UnixNano() internally applies additional truncation and uses both wall and ext time

Reply:
I think it's safe to remove this TODO and comment.

If users are setting precision of 1 μs this isn't even applicable in real-world systems, esp. given how hard it is to coordinate time in distributed systems. 

Tbh I'd even bump the tests to something like 1ms or even 10ms.